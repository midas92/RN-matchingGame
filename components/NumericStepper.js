import React from 'react';
import { View, StyleSheet, Text, TouchableOpacity } from 'react-native';

const NumericStepper = ({ 
  value,
  maxValue,
  minValue,
  stepDown = 1,
  stepUp = 1,
  onValueChange
}) => {
  return (
    <View style={styles.numericInputContainer}>
      <TouchableOpacity 
        style={{...styles.changeValueButton, ...styles.changeValueButtonMinus}}
        onPress={() => onValueChange(value - stepDown)}
        disabled={value == minValue}
      >
        <Text style={styles.changeValueText}>-</Text>
      </TouchableOpacity>

      <View style={styles.valueInput}>
        <Text style={styles.changeValueText}>{value}</Text>
      </View>
      
      <TouchableOpacity 
        style={{...styles.changeValueButton, ...styles.changeValueButtonPlus}}
        onPress={() => onValueChange(value + stepUp) }
        disabled={value == maxValue}
      >
        <Text style={styles.changeValueText}>+</Text>
      </TouchableOpacity>
    </View>
  );
}

export default NumericStepper;

const styles = StyleSheet.create({
  numericInputContainer: {
    flexDirection: 'row',
  },
   valueInput: {
     width: 50,
     height: 50,

     justifyContent: 'center',
     alignItems: 'center',

     borderColor: '#cccccc',
     borderWidth: 1
   },
   changeValueButton: {
    width: 50,
    height: 50,
    
    justifyContent: 'center',
    alignItems: 'center',

    borderColor: '#cccccc',
    borderWidth: 1
   },
   changeValueButtonPlus: {
    borderBottomRightRadius: 5,
    borderTopRightRadius: 5,
   },
   changeValueButtonMinus: {
    borderBottomLeftRadius: 5,
    borderTopLeftRadius: 5,
   },
   changeValueText: {
    fontSize: 20
   },
})