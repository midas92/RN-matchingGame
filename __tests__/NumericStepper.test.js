import React from 'react';
import { render } from '@testing-library/react-native';
import NumericStepper from '../components/NumericStepper';

describe('<NumericStepper/>', () => {
  it('should match snapshot', () => {
    const result = render(<NumericStepper/>).toJSON();
    expect(result).toMatchSnapshot();
  });

  it('should render the NumericStepper component', () => {
    const result = render(<NumericStepper/>);
    expect(result).toBeTruthy();
  });
});
