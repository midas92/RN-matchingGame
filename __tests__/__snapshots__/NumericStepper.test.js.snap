// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NumericStepper/> should match snapshot 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderBottomLeftRadius": 5,
        "borderColor": "#cccccc",
        "borderTopLeftRadius": 5,
        "borderWidth": 1,
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
        "width": 50,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    >
      -
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#cccccc",
        "borderWidth": 1,
        "height": 50,
        "justifyContent": "center",
        "width": 50,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderBottomRightRadius": 5,
        "borderColor": "#cccccc",
        "borderTopRightRadius": 5,
        "borderWidth": 1,
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
        "width": 50,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    >
      +
    </Text>
  </View>
</View>
`;
